src/python/library/tritonclient/utils/cuda_shared_memory/__init__.py

```diff
@@ -205,21 +172,31 @@ def set_shared_memory_region(cuda_shm_handle, input_values):
                 "input_values must be specified as a list/tuple of numpy arrays"
             )
 
+    _support_uva(cuda_shm_handle._device_id, -1)
+    stream = _get_or_create_global_cuda_stream()
+
     offset_current = 0
     for input_value in input_values:
         input_value = np.ascontiguousarray(input_value).flatten()
+        # When the input array is in type "BYTES" (np.object_ is
+        # the numpy equivalent), expect the array has been serialized
+        # via 'tritonclient.utils.serialize_byte_tensor'.
         if input_value.dtype == np.object_:
             input_value = input_value.item()
             byte_size = np.dtype(np.byte).itemsize * len(input_value)
-            _raise_if_error(
-                c_int(_ccudashm_shared_memory_region_set(cuda_shm_handle, c_uint64(offset_current), \
-                    c_uint64(byte_size), cast(input_value, c_void_p), -1)))
+            err = cudart.cudaMemcpyAsync(
+                cuda_shm_handle._base_addr + offset_current, input_value,
+                byte_size, cudart.cudaMemcpyKind.cudaMemcpyDefault, stream)
+            _raise_errno_if_cuda_err(err, -3)
         else:
             byte_size = input_value.size * input_value.itemsize
-            _raise_if_error(
-                c_int(_ccudashm_shared_memory_region_set(cuda_shm_handle, c_uint64(offset_current), \
-                    c_uint64(byte_size), input_value.ctypes.data_as(c_void_p), -1)))
+            err = cudart.cudaMemcpyAsync(
+                cuda_shm_handle._base_addr + offset_current,
+                input_value.ctypes.data, byte_size,
+                cudart.cudaMemcpyKind.cudaMemcpyDefault, stream)
+            _raise_errno_if_cuda_err(err, -3)
         offset_current += byte_size
+    _raise_errno_if_cuda_err(cudart.cudaStreamSynchronize(stream), -8)
```

Comment:
I was wondering if it is possible to get the error string from the error object instead of setting the error message and manually set the numbers/enum by ourselves.